binary_search_tree.rb

```diff
@@ -0,0 +1,209 @@
+require_relative "node"
+
+class BinarySearchTree
+  attr_accessor :root
+
+  def initialize()
+    @root = nil
+  end
+
+  def add_node(element)
+    new_node = Node.new(element)
+    if @root.nil?
+      @root = new_node
+    else
+      parent = nil
+      current = @root
+      until current.nil?
+        parent = current
+        if element < current.value
+          current = current.left_node
+        else
+          current = current.right_node
+        end
+      end
+      if element < parent.value
+        parent.left_node = new_node
+      else
+        parent.right_node = new_node
+      end
+    end
+  end
+
+  def add_elements(elements)
+    if elements.size
```

Comment:
I think this is not required, validated it. Also can elements be nil? If yes this will through error